401/src/main/java/graph/Graph.java

```diff
@@ -0,0 +1,65 @@
+package graph;
+
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public class Graph<T> {
+    private List<Node<T>> nodes;
+
+    //Creates a new Graph object with an empty ArrayList of Nodes
+    public Graph() {
+        this.nodes = new ArrayList<>();
+    }
+
+    //returns the ArrayList of Nodes for a graph
+    public List<Node<T>> getNodes() {
+        return nodes;
+    }
+
+    //Returns and integer representing how many Nodes are in the graph
+    public int size() {
+        return nodes.size();
+    }
+
+    //Takes in a value and adds a new Node with that value to the graph. Returns the new Node
+    public Node<T> addNode(T value) {
+        Node<T> n = new Node<>(value);
+        this.nodes.add(n);
+        return n;
+    }
+
+    //Takes in 2 nodes and adds an Edge to each one to create the relation that they are neighbors. The new Edges will have a weight of 0 with this method
+    public void addEdge(Node<T> node1, Node<T> node2) throws NoSuchElementException, InvalidParameterException {
```

Comment:
Since you have a lot of duplicated logic here, would be nicer if it just called `this.addEdge(node1, node2, 0)